core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/DistroXService.java

```diff
@@ -60,8 +60,8 @@ public StackV4Response post(DistroXV1Request request) {
     private void validate(DistroXV1Request request) {
         DetailedEnvironmentResponse environment = Optional.ofNullable(environmentClientService.getByName(request.getEnvironmentName()))
                 .orElseThrow(() -> new BadRequestException("No environment name provided hence unable to obtain some important data"));
-        if (environment == null) {
-            throw new BadRequestException(format("'%s' Environment does not exist.", request.getEnvironmentName()));
+        if (environment.getEnvironmentStatus().isDeleteInProgress()) {
+            throw new BadRequestException(format("'%s' Environment can not be delete in progress state.", request.getEnvironmentName()));
```

Comment:
Why blocking only this particular state? What about the rest that are not `isAvailable()` ones? For instance why permit DH creation if the env is stopped or stop-in-progress (`STOP*`, `ENV_STOPPED`)? I am not sure how to treat the various upgrade/update ones (e.g. `LOAD_BALANCER_ENV_UPDATE*`, `UPGRADE_CCM*`, `VERTICAL_SCALE*`, `PROXY_CONFIG_MODIFICATION*`).

Reply:
if we only enable the creation in case of avaialable state then I am afraid any env related action which does not disturb datahubs also blocking the whole workflow on an environment. I thought it is a better idea to focus on this particular problem.